### PR TITLE
AP_Logger: create and use AP_Logger_config.h

### PR DESCRIPTION
**Pull request by peterbarker**
This is just moving the config stuff into a separate file; should make no functional difference.

### CODE REVIEW

**Comment by peterbarker** [comment]
```
Board              AP_Periph  blimp  copter  heli  plane  rover  sub
Durandal                      0      0       0     0      0      0
Hitec-Airspeed     0                                             
KakuteH7-bdshot               0      0       0     0      0      0
MatekF405                     0      0       0     0      0      0
Pixhawk1-1M                   0      0       0     0      0      0
f103-QiotekPeriph  0                                             
f303-Universal     0                                             
revo-mini                     0      0       0     0      0      0
skyviper-v2450                       0                           
```

File: libraries/AP_Logger/AP_Logger_config.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL_Boards.h>
```
Comment (IamPete1) [inline libraries/AP_Logger/AP_Logger_config.h:3]: We gained this include? looks identical apart from that.

Comment (peterbarker) [reply libraries/AP_Logger/AP_Logger_config.h:3]: This *must* be included if you want your `#ifndef`s to work properly :-)   This is where the stuff from hwdef.h comes in, for example.

**Review by tridge — APPROVED** [review]

File: libraries/AP_Logger/AP_Logger_config.h (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_Logger/AP_Logger_config.h:3]: Does that mean they didn't work before?

**Pull request merged**